nucypher/blockchain/eth/clients.py

```diff
@@ -278,12 +376,15 @@ def sign_message(self, account: str, message: bytes) -> str:
         """
         return self.w3.eth.sign(account, data=message)
 
+    def get_blocktime(self):
+        highest_block = self.w3.eth.getBlock('latest')
+        now = highest_block['timestamp']
+        return now
+
     def _has_latest_block(self) -> bool:
+        # TODO: Investigate using `web3.middleware.make_stalecheck_middleware` #2060
```

Comment:
Can this entire receipt finality strategy be implemented as a web3 middleware?

Reply:
I think it may not be possible, since the number of confirmations is inputted by the user.